src/main/java/org/testng/TestRunner.java

```diff
@@ -961,7 +970,18 @@ public ISuite getSuite() {
 
   @Override
   public ITestNGMethod[] getAllTestMethods() {
-    return m_allTestMethods;
+    if (m_allJunitTestMethods != null) {
```

Comment:
What do you think if we use `m_xmlTest.isJUnit()` instead here?

Reply:
Fixed as recommended